[enhancement](backup) handle dropped tables and partitions during backup

peterylh opened the pull request:
### What problem does this PR solve?

Issue Number: close #52274 

Problem Summary:

When a backup is running, dropping a table, dropping a partition or truncating a table would cause backup failure because BE cannot find some tablets. This fix:

1. Add droppedTables and droppedPartitionsByTable fields to track dropped objects during backup
2.  Filter out dropped tables and partitions during backup

This ensures backup jobs can complete successfully even when tables or partitions are dropped during the backup process.
### Release note

None

### Check List (For Author)

- Test <!-- At least one of them must be included. -->
    - [X] Regression test
    - [X] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason <!-- Add your reason?  -->

- Behavior changed:
    - [X] No.
    - [ ] Yes. <!-- Explain the behavior change -->

- Does this need documentation?
    - [X] No.
    - [ ] Yes. <!-- Add document PR link here. eg: https://github.com/apache/doris-website/pull/1214 -->

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label <!-- Add branch pick label that this PR should merge into -->

peterylh committed:
[fix](backup) handle dropped tables and partitions during backup process (#52274)

When a backup is running, dropping a table, dropping a partition or truncating a table would cause backup failure because BE cannot find some tablets. This fix:

1. Add droppedTables and droppedPartitionsByTable fields to track dropped objects during backup
2.  Filter out dropped tables and partitions during backup

This ensures backup jobs can complete successfully even when tables or partitions are dropped during the backup process.

Thearas commented:
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

peterylh committed:
[fix](backup) handle case when all tables are dropped during backup preparation (#52274)

peterylh commented:
run buildall

dataroaring commented on fe/fe-core/src/main/java/org/apache/doris/backup/BackupJob.java:
```diff
@@ -813,6 +875,43 @@ private void waitingAllUploadingFinished() {
+                        removedTable.getName(), tableId, this);
+                backupMeta.removeTable(tableId);
+            }
+        }
```
droppedPartitionsByTable.clear();
droppedTables.clear();

because they are useless here to reduce editlog size.

peterylh committed:
[enhancement](backup) add comprehensive test cases and improve cleanup logic

Commit:
[fix](test) Fix code style violations in BackupJobTest

- Remove unused import MaterializedIndex
- Fix import order for Table class
- Remove trailing whitespace characters

This commit resolves checkstyle violations:
- UnusedImports error
- CustomImportOrder error
- RegexpSingleline errors for trailing whitespace

peterylh commented:
run buildall

dataroaring commented:
run buildall

peterylh committed:
Merge branch 'master' into b52274

peterylh committed:
[test](fe) update BackupContent reference in BackupJobTest

peterylh commented:
run buildall

peterylh commented:
run feut

hello-stephen commented:
# FE UT Coverage Report
Increment line coverage `  45.68% (37/81)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/52935_e52438c916a4215212f37cd599dce84da94fb7b6/fe_increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/52935_e52438c916a4215212f37cd599dce84da94fb7b6/fe_report/index.html)

dataroaring reviewed (approved):
LGTM
